applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -1185,19 +1236,29 @@ static int usb_init_next(void)
 		return -ENXIO;
 	}
 
+	if (!usbd_can_detect_vbus(usbd)) {
+		LOG_ERR("USBD controller cannot detect VBUS state change");
+		return -EIO;
+	}
+
 	err = usbd_msg_register_cb(usbd, usb_init_next_status_cb);
 	if (err) {
 		LOG_ERR("usbd_msg_register_cb failed (err: %d)", err);
 		return err;
 	}
 
+	if (IS_ENABLED(CONFIG_DESKTOP_USB_STACK_NEXT_DISABLE_ON_VBUS_REMOVAL)) {
```

Comment:
what if the cable is connected during the flash operation? Would it still work correctly?

Reply:
Yes, the callback is called after registration.

Reply:
Actually it's called after usbd_init so I will register it befot calling usbd_init - that way we have no races.